servicetalk-concurrent-test-internal/src/main/java/io/servicetalk/concurrent/test/internal/AwaitUtils.java

```diff
@@ -122,12 +122,35 @@ public static boolean awaitUninterruptibly(CountDownLatch latch, long timeout, T
         }
     }
 
+    /**
+     * {@link BlockingQueue#take()} from the queue or throws unchecked exception in the case
+     * of InterruptedException.
+     * @param queue The queue to take from.
+     * @param <T> The types of objects in the queue.
+     * @return see {@link BlockingQueue#take()}.
+     */
+    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings(
+            value = "NP_NONNULL_RETURN_VIOLATION",
+            justification = "return null statement is never reached"
+    )
+    public static <T> T take(BlockingQueue<T> queue) {
+        try {
+            return queue.take();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throwException(e);
+            return null;
```

Comment:
This is how we workarounded this issue in `PlatformDependent` class:
https://github.com/apple/servicetalk/blob/464554ad984bc4897e7ba72e54a2ac0f12a9ecd9/servicetalk-utils-internal/src/main/java/io/servicetalk/utils/internal/PlatformDependent.java#L163-L165

Consider following a similar approach and add a comment to clarify why this is necessary.

Reply:
My other solution was to return Object cast to T, but never got to the point to also include `@SuppressWarnings("unchecked")`. This indeed makes it more straightforward and readable, thank you :)